packages/block-library/src/comments/edit/comments-legacy.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	AlignmentControl,
+	BlockControls,
+	Warning,
+	useBlockProps,
+} from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
+import { Button } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import Placeholder from './placeholder';
+
+export default function CommentsLegacy( {
+	attributes,
+	setAttributes,
+	context: { postType, postId },
+} ) {
+	const { textAlign } = attributes;
+
+	const removeLegacy = () => {
+		setAttributes( { legacy: false } );
+	};
+
+	const actions = [
+		<Button key="convert" onClick={ removeLegacy } variant="primary">
+			{ __( 'Switch to editable mode' ) }
+		</Button>,
+	];
+
+	const blockProps = useBlockProps( {
+		className: classnames( {
+			[ `has-text-align-${ textAlign }` ]: textAlign,
+		} ),
+	} );
+
+	return (
+		<>
+			<BlockControls group="block">
+				<AlignmentControl
+					value={ textAlign }
+					onChange={ ( nextAlign ) => {
+						setAttributes( { textAlign: nextAlign } );
+					} }
+				/>
+			</BlockControls>
+
+			<div { ...blockProps }>
+				<Warning actions={ actions }>
+					{ __(
+						"Comments block: You're currently using this block in legacy mode. " +
+							'The following is just a placeholder, not a real comment. ' +
+							'The final styling may differ because it also depends on the current theme. ' +
+							'For better compatibility with the Block Editor, ' +
+							'please consider switching the block to its editable mode.'
+					) }
```

Comment:
I _think_ `__()` (or rather, our string extraction util) is smart enough to handle basic string literal concatenation. See e.g. [this test](https://github.com/WordPress/gutenberg/blob/1a33c0d87b4da3898216f6d7a300e4af88c6d6ca/packages/babel-plugin-makepot/test/index.js#L137-L143).